optional/script/deploy

```diff
@@ -18,12 +18,12 @@ if [[ ! -f Staticfile.auth ]]; then
 fi
 echo "OK!"
 
-# Check that we're on master, or there's an environment variable set to override
+# Check that we're on main, or there's an environment variable set to override
 BRANCH=$(git rev-parse --abbrev-ref HEAD)
 
 echo "Checking branch..."
-if [[ $BRANCH != "master" ]] && [[ $I_REALISE_I_AM_NOT_ON_MASTER != "y" ]]; then
-  error "Current branch is not master, cowardly refusing to deploy"
+if [[ $BRANCH != "main" ]] && [[ $I_REALISE_I_AM_NOT_ON_MASTER != "y" ]]; then
+  error "Current branch is not main, cowardly refusing to deploy"
   error "To force override, run this again with I_REALISE_I_AM_NOT_ON_MASTER=y"
   exit 1
```

Comment:
I think the 'override' environment variable also needs updating?

```suggestion
if [[ $BRANCH != "main" ]] && [[ $I_REALISE_I_AM_NOT_ON_MAIN != "y" ]]; then
  error "Current branch is not main, cowardly refusing to deploy"
  error "To force override, run this again with I_REALISE_I_AM_NOT_ON_MAIN=y"
  exit 1
```

Reply:
Hmmm, good point... I wish I understood though what this is here for or if anyone uses it... this feels like it could be a breaking change, especially the way that it interacts with our expectations for the default branch name in the site configuration :/

Reply:
I think the idea was to allow for deploys from local machines (i.e. for those not using Travis etc – at the time GitHub Actions wasn't even a thing, I think) and to have this here as a 'safety rail' but give users the ability to override it if they knew what they were doing.

More generally, this repo is only ever really relevant when [creating a new documentation project](https://tdt-documentation.london.cloudapps.digital/create_project/create_new_project/#create-a-new-documentation-project) which uses this repo as a [Middleman Project Template](https://middlemanapp.com/advanced/project-templates/).

At this point the Thorfile is used to ask the user a bunch of questions, including if they're planning to deploy to PaaS.

If they say yes, we copy the script/deploy folder into their documentation project:

https://github.com/alphagov/tech-docs-template/blob/ff2013c81ea630ca487e45f817c4105a522651d3/Thorfile#L98-L106

After that, when updating their project, we [tell them to update the gem instead](https://tdt-documentation.london.cloudapps.digital/maintain_project/use_latest_template/). (Before the gem existed, we got them to do something else clever with used Project Templates which I can't remember at this point)

Basically this shouldn't affect anyone updating, only new projects, so we should be relatively free to make other changes to this file if we needed to.

(Given PaaS is being decommissioned, we should possibly just simplify and remove anything to do with deployment from this template anyway at some point?)

Reply:
Ah, thank you so much for the context, that's really helpful.

> (Given PaaS is being decommissioned, we should possibly just simplify and remove anything to do with deployment from this template anyway at some point?)

Yeah, good point :'(

Reply:
More background on the move to a gem here, in case it's interesting / useful:
https://github.com/alphagov/tech-docs-template/issues/126

Reply:
The fact it's only used for new projects makes me feel more comfortable about changing it, I am now leaning towards making the change in tech-docs-gem to use 'main' even if it is breaking, and just tell users to figure it out 😅